viewing_party/party.py

```diff
@@ -1,23 +1,119 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    inputs = [title, genre, rating]
+    for input in inputs:
+        if input == None:
+            return None
+    new_movie = {}
+    new_movie["title"] = title
+    new_movie["genre"] = genre
+    new_movie["rating"] = rating
+    return new_movie
+
+def add_to_watched(user_data, movie): 
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, movie_title):
+    movie_index = 0
+    for movie in user_data["watchlist"]:
+        if movie_title == movie["title"]:
+            user_data["watched"].append(movie)
+            user_data["watchlist"].pop(movie_index)
+        movie_index += 1
+
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
-
+def get_watched_avg_rating(user_data):
+    if len(user_data["watched"]) == 0:
+        return 0.0
+    ratings_list = []
+    for movie in user_data["watched"]:
+        ratings_list.append(movie["rating"])
+    return sum(ratings_list)/len(ratings_list)
+    
+def get_most_watched_genre(user_data):
+    if len(user_data["watched"]) == 0:
+        return None
+    genre_count = {}
+    for movie in user_data["watched"]:
+        movie_genre = movie["genre"]
+        if movie_genre not in genre_count:
+            genre_count[movie_genre] = 1
+        elif movie_genre in genre_count:
+            genre_count[movie_genre] += 1
+    
+    genre_max = max(genre_count, key=genre_count.get)

```

Comment:
Nice use of max here. I find it useful to also practice writing your own implementation of finding a max value from a dictionary. You may find in an interview if you use built in functions like max that an interviewer may ask you how the function works under the hood. So knowing how to write it on your own can help you better understand the function.